Framework/CurveFitting/src/Functions/CrystalFieldPeaksBase.cpp

```diff
@@ -336,11 +336,37 @@ void CrystalFieldPeaksBase::calculateEigenSystem(DoubleFortranVector &en,
 
   auto ionIter = ION_2_NRE.find(ion);
   if (ionIter == ION_2_NRE.end()) {
-    throw std::runtime_error("Unknown ion name passed to CrystalFieldPeaks.");
+    // If 'Ion=S2', or 'Ion=J2.5' etc, interpret as arbitrary J values with gJ=2
+    // Allow lower case, but values must be half-integral. E.g. 'Ion=S2.4' fails
+    switch (ion[0]) {
+    case 'S':
+    case 's':
+    case 'J':
+    case 'j': {
+      std::string numbers("0123456789");
+      if (numbers.find(ion[1]) != std::string::npos) {
+        // Need to store as 2J to allow half-integer values
+        try {
+          auto J2 = std::stof(ion.substr(1)) * 2.;
+          if (fabs(J2 - (int)J2) < 0.001) {
+            nre = -(int)J2;
+            break;
+          }
+          // Catch exceptions thrown by stof so we get a more meaningful error
+        } catch (const std::invalid_argument &) {
+        } catch (const std::out_of_range &) {
+        }
+      }
```

Comment:
Maybe an exception with a different message should be thrown here warning that the `J` value is wrong